tests/pytests/unit/spm/test_spm.py

```diff
@@ -0,0 +1,211 @@
+    fdir = str(_tmp_spm / formula["definition"]["name"])
+    shutil.rmtree(fdir, ignore_errors=True)
+    os.mkdir(fdir)
+    for path, contents in formula["contents"]:
+        path = os.path.join(fdir, path)
+        dirname, _ = os.path.split(path)
+        if not os.path.exists(dirname):
+            os.makedirs(dirname)
+        with salt.utils.files.fopen(path, "w") as f:
+            f.write(contents)
+    return fdir
+
+
+def test_build_install(setup_spm, f1_content):
+    # Build package
+    _tmp_spm, ui, client, minion_config, minion_opts = setup_spm
+    fdir = _create_formula_files(f1_content, _tmp_spm)
+    with patch("salt.client.Caller", MagicMock(return_value=minion_opts)):
+        with patch(
+            "salt.client.get_local_client",
+            MagicMock(return_value=minion_opts["conf_file"]),
+        ):
+            client.run(["build", fdir])
+    pkgpath = ui._status[-1].split()[-1]
+    assert os.path.exists(pkgpath)
+    # Install package
+    with patch("salt.client.Caller", MagicMock(return_value=minion_opts)):
+        with patch(
+            "salt.client.get_local_client",
+            MagicMock(return_value=minion_opts["conf_file"]),
+        ):
+            client.run(["local", "install", pkgpath])
+    # Check filesystem
+    for path, contents in f1_content["contents"]:
+        path = os.path.join(
+            minion_config["file_roots"]["base"][0],
+            f1_content["definition"]["name"],
+            path,
+        )
+        assert os.path.exists(path)
+        with salt.utils.files.fopen(path, "r") as rfh:
+            assert rfh.read() == contents
+    # Check database
+    with patch("salt.client.Caller", MagicMock(return_value=minion_opts)):
+        with patch(
+            "salt.client.get_local_client",
+            MagicMock(return_value=minion_opts["conf_file"]),
+        ):
+            client.run(["info", f1_content["definition"]["name"]])
+    lines = ui._status[-1].split("\n")
+    for key, line in (
+        ("name", "Name: {0}"),
+        ("version", "Version: {0}"),
+        ("release", "Release: {0}"),
+        ("summary", "Summary: {0}"),
+    ):
+        assert line.format(f1_content["definition"][key]) in lines
+    # Reinstall with force=False, should fail
+    ui._error = []
+    with patch("salt.client.Caller", MagicMock(return_value=minion_opts)):
+        with patch(
+            "salt.client.get_local_client",
+            MagicMock(return_value=minion_opts["conf_file"]),
+        ):
+            client.run(["local", "install", pkgpath])
+    assert len(ui._error) > 0
+    # Reinstall with force=True, should succeed
+    with patch.dict(minion_config, {"force": True}):
+        ui._error = []
+        with patch("salt.client.Caller", MagicMock(return_value=minion_opts)):
+            with patch(
+                "salt.client.get_local_client",
+                MagicMock(return_value=minion_opts["conf_file"]),
+            ):
```

Comment:
**question** it looks like this patch pair happens before every call - there's one place where I notice that the minion config gets changed, but otherwise it's all the same. Could these patches just be moved to a fixture?